### DISCUSSION TITLE
Color Picker Widget plugin

### DISCUSSION THREAD

**Discussion by curtqikfox** [Milkdown/milkdown#1234] [Q&A]
Hello, I am using an older version of milkdown (6.5.4) in order to gain access to the deprecated plugin @milkdown/plugin-menu. I am trying to figure out how I can create a plugin for a color picker wheel on the plugin-menu, and allows the user to highlight some text in the editor, and change its color.

Please let me know if this is possible, or if there is a newer way to create and manage a toolbar that does not use the @milkdown/plugin-menu plugin.

**Comment by zgrybus** [comment]
@curtqikfox The best way to archive that is I believe:
1. Upgrading to the newest version od Milkdown.
2. Creating custom highlight by colour mark.
3. Use any lib that provides `colour picker wheel`.
4. When user select any color, it should highlight selected text ( right? ). If yes, then simply replace selection with your mark and add appropriate `attrs` to it. - https://github.com/DareData/notion-style-editor/blob/main/src/lib/hooks/useEditorLinkActions.tsx#L18
You may inspire from this repo that I've created :) 
https://github.com/DareData/notion-style-editor
